server/src/test/java/org/elasticsearch/index/query/TypeQueryBuilderTests.java

```diff
@@ -75,12 +75,18 @@ public void testFromJson() throws IOException {
     @Override
     public void testToQuery() throws IOException {
         super.testToQuery();
-        assertWarnings("The [type] query is deprecated, filter on a field instead.");
+        assertWarnings(
+            "The [type] query is deprecated, filter on a field instead.",
```

Comment:
Maybe we could avoid these duplicate warning messages? One idea:
- We generalize the message in `TypeFieldType.TYPES_DEPRECATION_MESSAGE` to say "[types removal] Referring to types within search queries is deprecated, filter on a field instead."
- We make sure to use `deprecatedAndMaybeLog` in `TypeQueryBuilder` to avoid producing the warning twice.